app/services/update_auction.rb

```diff
@@ -1,20 +1,40 @@
-class UpdateAuction < Struct.new(:auction, :params, :user)
+class UpdateAuction
+  def initialize(auction, params)
+    @auction = auction
+    @params = params
+  end
+
   def perform
     auction.assign_attributes(attributes)
+    create_purchase_request
```

Comment:
Do you want to have something up here where it returns if the cap_request_url has been set?